base/libgit2/types.jl

```diff
@@ -177,22 +177,44 @@ function RemoteCallbacks{P<:AbstractPayload}(credentials::Ptr{Void}, payload::Nu
     end
 end
 
-immutable FetchOptions
-    version::Cuint
-    callbacks::RemoteCallbacks
-    prune::Cint
-    update_fetchhead::Cint
-    download_tags::Cint
-end
-FetchOptions(; callbacks::RemoteCallbacks = RemoteCallbacks(),
-               prune::Cint = Consts.FETCH_PRUNE_UNSPECIFIED,
-               update_fetchhead::Cint = one(Cint),
-               download_tags::Cint = Consts.REMOTE_DOWNLOAD_TAGS_AUTO
-) = FetchOptions(one(Cuint),
-                 callbacks,
-                 prune,
-                 update_fetchhead,
-                 download_tags)
+if LibGit2.version() >= v"0.24.0"
+    immutable FetchOptions
+        version::Cuint
+        callbacks::RemoteCallbacks
+        prune::Cint
+        update_fetchhead::Cint
+        download_tags::Cint
+        custom_headers::StrArrayStruct
```

Comment:
Actually, `CloneOptions` has a `FetchOptions` field in the middle, so the size of the struct actually matters. I wouldn't have spotted this change (not mentioned in the NEWS) if it wasn't for that.